src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1014 @@
+
+    # Step 6: Construct relinquishment request for grant 'G1'.
+    relq_request = {
+        'relinquishmentRequest': [{
+            'cbsdId': cbsd_ids[0],
+            'grantId': grant_response_g1[0]['grantId']
+        }]
+    }
+
+    # Send relinquishment request for grant 'G1'.
+    relq_response = self._sas.Relinquishment(relq_request)['relinquishmentResponse']
+    self.assertEqual(relq_response[0]['response']['responseCode'], 0)
+
+    del relq_request, relq_response
+
+    # TOD0
+    # Step 7: Modify the GWBL database to include a modified version of the GWBL 'W'.
+    # 'W' is moved further than 150 kms from FSS site.
+
+    # Step 8: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 9: Request grant 'G2'.
+    config['grantRequests'][1]['cbsdId'] = cbsd_ids[0]
+
+    grant_request_g2 = {'grantRequest': [config['grantRequests'][1]]}
+    grant_response_g2 = self._sas.Grant(grant_request_g2)['grantResponse']
+
+    # Check the grant response code is as expected (SUCCESS or INTERFERENCE).
+    self.assertIn(grant_response_g2[0]['response']['responseCode'],
+                      config['expectedResponseCodes'][1])
+
+  def generate_FDB_7_default_config(self, filename):
+    """Generates the WinnForum configuration for FDB.7"""
+
+    # Load device info
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Update the FCC ID of CBSD 'C' with the FCC ID 'F_ID' from FCC ID database.
+    device_b['fccId'] = "F_ID"
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequest': [device_b],
+        'expectedResponseCodes': [(103,), (0,)],
+        'conditionalRegistrationData': conditionals
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FDB_7_default_config)
+  def test_WINNF_FT_S_FDB_7(self, config_filename):
+    """DPA Database Update for DPAs which cannot be monitored by an ESC."""
+
+    # Load the configuration file
+    config = loadConfig(config_filename)
+
+    registration_request = {
+        'registrationRequest': config['registrationRequest']
+    }
+
+    # Step 1: Inject FCC ID and User Id of the devices into UUT.
+    self._sas_admin.InjectFccId({'fccId': registration_request['registrationRequest'][0]['fccId']})
```

Comment:
You can't inject the FCC IDs: the whole point of the test case is to see that the SAS UUT correctly pulls this information from the DB.

Reply:
Updated,   Removed the injection of FCC ID and User ID.